docs/development_testing.md

```diff
@@ -84,10 +84,8 @@ pre-commit run --show-diff-on-failure
 The linters are also available directly, you can run tests on individual files:
 
 ```shell
-flake8 homeassistant/core.py
 pylint homeassistant/core.py
 black homeassistant/core.py
-isort homeassistant/core.py
```

Comment:
As these were replaced by ruff, we should definitely add it here. Adding other linters to the list sounds also good to me.

Reply:
I wonder if it makes sense to even recommend running these linters manually?

From what I can tell, `pre-commit` seems to be the source of truth and if you run it with the example just above this snippet in the code, or even with something like `pre-commit run --files homeassistant/components/my_component/*`, it's pretty fast.

Otherwise we risk having to update this list every time we add or remove linters.

Reply:
Agreed, simpler is better and TIL about `--files`, thanks! I suppose the idea of running each linter separately comes from a time when running all linters and formatters against the whole code-base was slow? I know pylint still is, but if it works fast and as expected using `--files`, I see no reason not to go for suggesting using `pre-commit --show-diff-on-failure --files <placeholder>` to speed-up the checks.

Reply:
> I wonder if it makes sense to even recommend running these linters manually?

Ruff is fast, I just run `ruff .`, takes a second on the whole codebase 🤷

Reply:
Haha just tried it out myself, you're right! Should we just replace this whole block with that then?

Reply:
> Haha just tried it out myself, you're right! 

Hehe there is a reason why it is getting so much traction 🚀 

> Should we just replace this whole block with that then?

Only for the tools that are replaced by it I would say.

Reply:
Thanks, added a ruff example using the same format as the other ones (running it against `homeassistant/core.py`)